src/v3/src/hooks/useOnSubmit.ts

```diff
@@ -54,7 +58,14 @@ export const useOnSubmit = (): (options: OnSubmitHandlerOptions) => Promise<void
     if (params) {
       payload = merge(payload, params);
     }
-    payload = { ...payload, step };
+    if (isActionStep) {
+      payload = {
+        ...payload,
```

Comment:
It looks like `payload` includes `data` shouldn't that be included in the `params` part of the action? IIRC the only thing expected at the base of the object is stateHandle when there are actions, I thought everything else went into the action object (name and params for that action).

Reply:
upstream transformation logic controls that

Reply:
Sorry, i am referring to this file

```
if (includeData) {
      payload = merge(payload, omit(data, OMITTED_PROPERTIES));
    }
    if (params) {
      payload = merge(payload, params);
    }
```
Data is added to payload, then params are merged to payload but for actions, only params are passed. but payload contains field level data. Was this intentional? IIRC, any corresponding data should also be passed to params for the action. Am I mistaken?

Reply:
that what I mean by controlled by upstream, `includeData` is passed down from transformation layer -> ui component -> hook